src/ansible_navigator/configuration_subsystem/navigator_post_processor.py

```diff
@@ -66,31 +72,78 @@ class VolumeMountOption(Enum):
     z = "z"  # pylint: disable=invalid-name
 
 
+V = TypeVar("V", bound="VolumeMount")  # pylint: disable=invalid-name
+
+
+class VolumeMountError(Exception):
+    """Custom exception raised when building VolumeMounts."""
+
+
 @dataclass
 class VolumeMount:
     """Describes EE volume mounts."""
 
-    #: The name of the config option requiring this volume mount.
-    calling_option: str
-
-    #: The source path of the volume mount.
-    src: str
-
-    #: The destination path in the container for the volume mount.
-    dest: str
+    fs_source: str
+    """The source file system path of the volume mount"""
+    fs_destination: str
+    """The destination file system path in the container for the volume mount"""
+    settings_entry: str
+    """The name of the settings entry requiring this volume mount"""
+    source: C
+    """The settings source for this volume mount"""
+    options_string: str = ""
```

Comment:
The only way I can think of is to not use a dataclass but a generic class instead, and pass the string into the __init__.....

I think it would be messy compared to this.

Reply:
Can't we convert it before we call the constructor at all?

```python
options: Set[VolumeMountOption] = VolumeMount.options_from_str(options_str)
VolumeMount(src, dest, options)
```